stl/inc/syncstream

```diff
@@ -0,0 +1,330 @@
+// syncstream standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SYNCSTREAM_
+#define _SYNCSTREAM_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+#define __cpp_lib_syncbuf // Remove before commit, visual
+#ifndef __cpp_lib_syncbuf
+#pragma message("The contents of <syncstream> are available only with C++20 syncbuf support.")
+#else // ^^^ !defined(__cpp_lib_syncbuf) / defined(__cpp_lib_syncbuf) vvv
+#include <map>
+#include <memory>
+#include <mutex>
+#include <streambuf>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#pragma push_macro("emit")
+#undef new
+#undef emit
+
+_STD_BEGIN
+struct _Basic_syncbuf_global_mutex {
+private:
+    struct _Mutex_count_pair {
+        mutex _Mutex;
+        uint64_t _Ref_count = 0;
+    };
+    static map<void*, _Mutex_count_pair>& _Get_mutex_map() noexcept {
+        static map<void*, _Mutex_count_pair> _Mutex_Map;
+        return _Mutex_Map;
+    }
+    static mutex& _Get_map_mutex() noexcept {
+        static mutex _Mutex;
+        return _Mutex;
+    }
+    static mutex& _Get_mutex_for_instance(void* _Ptr) {
+        mutex& _Mutex = _Get_map_mutex();

```

Comment:
This variable is only used to initialize `scoped_lock`. It should be inlined. Ditto below.